File: vue-app/src/classes/HoofCheck.ts
import { HoofCheckStatuses } from "@/enum/HoofCheckStates";
import { IHoofCheck } from "@/interfaces/IHoofCheck";

export class HoofCheck implements IHoofCheck {
  frontLeft: HoofCheckStatuses;
  frontLeftTask: string;
  frontRight: HoofCheckStatuses;
  frontRightTask: string;
  backLeft: HoofCheckStatuses;
  backLeftTask: string;
  backRight: HoofCheckStatuses;
  backRightTask: string;

  constructor() {
    this.frontLeft = HoofCheckStatuses.Neutral;
    this.frontLeftTask = "";
    this.frontRight = HoofCheckStatuses.Neutral;
    this.frontRightTask = "";
    this.backLeft = HoofCheckStatuses.Neutral;
    this.backLeftTask = "";
    this.backRight = HoofCheckStatuses.Neutral;
    this.backRightTask = "";
  }
}

File: WebApi/Classes/HoofCheck.cs
public class HoofCheck
{
    public string frontLeft { get; set; } = HoofCheckStatuses.Neutral.ToString();
    public string frontLeftTask { get; set; } = string.Empty;
    public string frontRight { get; set; } = HoofCheckStatuses.Neutral.ToString();
    public string frontRightTask { get; set; } = string.Empty;
    public string backLeft { get; set; } = HoofCheckStatuses.Neutral.ToString();
    public string backLeftTask { get; set; } = string.Empty;
    public string backRight { get; set; } = HoofCheckStatuses.Neutral.ToString();
    public string backRightTask { get; set; } = string.Empty;
}
